### PR TITLE
7235 applications add

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #7235.  I have code ready for edit functionality, but wanted to keep this PR small to expedite a review for it.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION

application add edit

### CODE REVIEW

**Comment by AlexSCorey** [comment]
recheck

**Comment by nixocio** [comment]
I got this code locally. Changes seem to be working fine.

**Review by nixocio — APPROVED** [review]

File: awx/ui_next/src/screens/Application/ApplicationAdd/ApplicationAdd.test.jsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      initialEntries: ['/applications/add'],
+    });
+    ApplicationsAPI.readOptions.mockResolvedValue(options);
+
+    ApplicationsAPI.create.mockResolvedValue({ data: { id: 8 } });
+    await act(async () => {
+      wrapper = mountWithContexts(<ApplicationAdd />, {
+        context: { router: { history } },
+      });
+      await act(async () => {
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Application/ApplicationAdd/ApplicationAdd.test.jsx]: should these `act` calls be nested? Never seen that before.

Comment (AlexSCorey) [reply awx/ui_next/src/screens/Application/ApplicationAdd/ApplicationAdd.test.jsx]: No, they shoouldn't be

**Review by jakemcdermott — APPROVED** [review]

**Comment by dsesami** [comment]
Can we fix something that existed in the old UI but was a pain to deal with while we're here?

1. Set auth type, etc.
2. Set the client type to public, but leave the redirect URIs blank
3. Try to save.

an error pops up telling you that you need a redirect URI. Can we trigger the required * asterisk when this situation is set up, so the user knows to set it beforehand? Also, the form should highlight that field red when this error happens.

Also, can we capitalize "Redirect URIs"? (currently it's "Redirect uris")

**Review by dsesami — APPROVED** [review]
Perfect!

**Pull request merged**